jgiven-core/src/test/java/com/tngtech/jgiven/impl/ScenarioModelBuilderTest.java

```diff
@@ -398,6 +398,44 @@ public void the_Description_annotation_on_intro_words_is_evaluated() throws Thro
         assertThat( step.getWords().get( 0 ).getValue() ).isEqualTo( "another description" );
     }
 
+    @Test
+    public void filler_words_are_prepended_to_stage_methods() throws Throwable {
+        startScenario( "Scenario with filler words" );
+        given().there().is().something();
+        getScenario().finished();
+        StepModel step = getScenario().getScenarioCaseModel().getFirstStep();
+        assertThat( step.getCompleteSentence() ).isEqualTo( "Given there is something" );
+    }
+
+    @Test
+    public void the_Description_annotation_on_filler_words_is_evaluated() throws Throwable {
+        startScenario( "Scenario with @As annotation on filler words" );
+        given().a().filler_word_with_an_as_annotation().and_with().something_else();
+        getScenario().finished();
+        StepModel step = getScenario().getScenarioCaseModel().getFirstStep();
+        assertThat( step.getCompleteSentence() ).isEqualTo( "Given a Filler Word and with something else" );
+    }
+
+    @Test
+    public void punctuation_can_be_added_to_filler_words() throws Throwable {
+        startScenario( "Scenario with punctuation" );
+        given().there().is().something_filled().comma().another().something_filled().and_with().something_else();
+        getScenario().finished();
+        StepModel step = getScenario().getScenarioCaseModel().getFirstStep();
+        assertThat( step.getCompleteSentence() )
+            .isEqualTo( "Given there is something filled, another something filled and with something else" );
+    }
+
+    @Test
+    public void punctuation_can_be_used_at_the_start_of_a_sentence() throws Throwable {
+        startScenario( "Scenario with punctuation prefixing a sentence" );
+        given().something()
+        .hyphen().something_else();
```

Comment:
A so that's why you wanted to be able to add punctuation at the start of a sentence. 
However, even if we want to allow that. I'd still argue that handling of punctuation should be consistent. i.e. it always be part of another word, or never.

Reply:
If we observe punctuation without the use of filler words what is the desired behaviour? Ignore the punctuation and log a warning?

Reply:
Just a thought. How about scrapping `@Punctuation` altogether and having an attribute on `@FillerWord` which allows you to mark it as punctuation, obviously defaulting to false.

If you want to start a sentence with a punctuation filler you may.

Reply:
I like that idea. Functionally punctuation and filler words are already very similar...

Reply:
coming to think of it, maybe you want to have that option for every word... then you can do all kinds of crazy sentence building... However, there is no need to get to such a thing right away.

Reply:
Yes, my thinking exactly. I'll sort that out.

Reply:
> coming to think of it, maybe you want to have that option for every word... then you can do all kinds of crazy sentence building... However, there is no need to get to such a thing right away.

Just image the power of that combined with Navigable Stages. Oh yes!

Reply:
@l-1squared I have now incorporated the punctuation concept into filler words. I'm definitely happier with it now. What do you think?